### PR TITLE
Save token network graph on NoRouteFound errors

### PR DESCRIPTION
**Pull request by karlb**
This should make it easier to debug cases where we should not get
NoRouteFound.

### CODE REVIEW

File: src/pathfinding_service/model/token_network.py [hunk 1]
```diff
@@ -458,6 +459,40 @@ def check_path_request_errors(
...
+                        "{} [{}]\n".format(
+                            to_checksum_address(node),
+                            ""
+                            if address_to_reachability[node] == AddressReachability.REACHABLE
+                            else "style=filled",
+                        )
+                    )
+            for edge_attr in self.G.edges.values():
+                f.write(
+                    '"{}" -> "{}" [label={}]\n'.format(
```
Comment (palango) [inline src/pathfinding_service/model/token_network.py]: Does graphviz handle bidirectional graphs?

File: src/pathfinding_service/api.py [hunk 2]
```diff
@@ -117,6 +118,13 @@ def __init__(self, debug_mode: bool, **kwargs: Any):
         self.debug_mode = debug_mode
 
     def post(self, token_network_address: str) -> Tuple[dict, int]:
+        graph_dir: Optional[str]
+        if self.debug_mode:
+            graph_dir = "/tmp/pfs_graphs"  # TODO: make this configurable
```
Comment (palango) [inline src/pathfinding_service/api.py]: Do we need to mount `/tmp` in the docer containers?

File: src/pathfinding_service/model/token_network.py (hunk 1 above)
Comment (karlb) [reply src/pathfinding_service/model/token_network.py]: I've modeled it as a directed graph with a pair of edges between the nodes and that works.

File: src/pathfinding_service/api.py (hunk 2 above)
Comment (karlb) [reply src/pathfinding_service/api.py]: I don't think we should do that. We should make that path configurable and do a bind mount for it. Until then we'll have fetch it from inside the container. Something like `docker run "cat /tmp/pfs_graphs/*.log"` should do it.

Comment (karlb) [reply src/pathfinding_service/api.py]: We could also bind `/tmp/pfs_graphs`, but really think binding all of `/tmp` is a bad idea.

Comment (palango) [reply src/pathfinding_service/api.py]: Ah cool, I didn't know about `docker run "cat /tmp/pfs_graphs/*.log"`.

**Comment by konradkonrad** [comment]
What's up with this PR @karlb ? Do you still want to get this merged?

**Comment by karlb** [comment]
> What's up with this PR @karlb ? Do you still want to get this merged?

Looks like all path problems could be explained with "presence is messed up", so far. So this seems to be not necessary. I'll just close it.

**Pull request closed without merging**